src/openms/include/OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimator.h

```diff
@@ -204,12 +181,9 @@ namespace OpenMS
     //MEMBERS:
 
     /// stores the noise estimate for each peak
-    std::map<PeakType, double, typename PeakType::PositionLess> stn_estimates_;
-
-    /// points to the first raw data point in the interval
-    PeakIterator first_;
-    /// points to the right position next to the last raw data point in the interval
-    PeakIterator last_;
+    std::vector<double> stn_estimates_;
+    /// raw data;
+    Container c;
```

Comment:
this seems very wasteful to copy the whole underlying data!

Can you try and remove this member entirely?!
The user calls .init(container) and there we can compute the S/N values and store them in `stn_estimates`. There should be no need to store the container as well.
The only thing that does not work anymore is the `is_result_valid_` idea. But that should not really be required and can be removed as well.

This should make the code quite a bit faster...

Reply:
looking at the code, you can simply remove the `is_result_valid_` member, and within `SignalToNoiseEstimator::update()` simply do `stn_estimates_.clear();`, this way, if someone sets new parameters and forgets to call .init() afterwards, we will see a segfault and thus the error is clear. You can also add a OPENMS_PRECONDITION to check the index is within the vectors size in `getSignalToNoise()`, so in debug builds we will get a proper error message and not just a segfault.